provider/provider.go

```diff
@@ -785,42 +786,64 @@ func (p *Provider) DispatchWithCtx(ctx context.Context, appPublicKey, chain stri
 }
 
 // Relay does request to be relayed to a target blockchain
-func (p *Provider) Relay(rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, error) {
+func (p *Provider) Relay(rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, *RelayOutputErr) {
 	return p.RelayWithCtx(context.Background(), rpcURL, input, options)
 }
 
 // RelayWithCtx does request to be relayed to a target blockchain
-func (p *Provider) RelayWithCtx(ctx context.Context, rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, error) {
+func (p *Provider) RelayWithCtx(ctx context.Context, rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, *RelayOutputErr) {
 	rawOutput, reqErr := p.doPostRequest(ctx, rpcURL, input, ClientRelayRoute, http.Header{})
 
 	defer closeOrLog(rawOutput)
 
 	if reqErr != nil && !errors.Is(reqErr, errOnRelayRequest) {
-		return nil, reqErr
+		return nil, &RelayOutputErr{Error: reqErr, StatusCode: rawOutput.Status}
 	}
 
-	bodyBytes, err := ioutil.ReadAll(rawOutput.Body)
+	bodyBytes, err := io.ReadAll(rawOutput.Body)
 	if err != nil {
-		return nil, err
+		return nil, &RelayOutputErr{Error: err, StatusCode: rawOutput.Status}
 	}
 
 	if errors.Is(reqErr, errOnRelayRequest) {
-		return nil, parseRelayErrorOutput(bodyBytes, input.Proof.ServicerPubKey)
+		return nil, &RelayOutputErr{Error: parseRelayErrorOutput(bodyBytes, input.Proof.ServicerPubKey), StatusCode: rawOutput.Status}
 	}
 
 	return parseRelaySuccesfulOutput(bodyBytes)
 }
 
-func parseRelaySuccesfulOutput(bodyBytes []byte) (*RelayOutput, error) {
+// TODO: REMOVE THIS HACKY FUNCTION
```

Comment:
Is this an actual TODO? 

1. If so, can you leave a comment on how you're thinking about removing it and why?
2. If not, add a godoc explaining what this function is.

I'm personally unsure (off the top off my head) what the better solution is.

Reply:
I can update the comment. The idea is that the node respond back with the status for REST chains, so the node must get back to us with response, signature and statusCode

Reply:
Providing this level of context in the comment is perfect!